### ISSUE TITLE
[java] SingletonClassReturningNewInstance false positive due to type cast

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#5158]
**Affects PMD Version:** 7.4.0

**Rule:** https://docs.pmd-code.org/latest/pmd_rules_java_errorprone.html#singletonclassreturningnewinstance

**Description:**

**Code Sample demonstrating the issue:**

```java
public final class ObjectArrayValidatorNoOp 
{
	private static final ObjectArrayValidatorNoOp INSTANCE = new ObjectArrayValidatorNoOp();

	public static <T, E> ObjectArrayValidator<T, E> getInstance()
	{
		@SuppressWarnings("unchecked")
		ObjectArrayValidator<T, E> temp = (ObjectArrayValidator<T, E>) INSTANCE;
		return temp;
	}
}
```

**Expected outcome:**

Equivalent output (no error) as when this code is used:

```java
public final class ObjectArrayValidatorNoOp 
{
	private static final ObjectArrayValidatorNoOp INSTANCE = new ObjectArrayValidatorNoOp();

	@SuppressWarnings("unchecked")
	public static <T, E> ObjectArrayValidator<T, E> getInstance()
	{
		return (ObjectArrayValidator<T, E>) INSTANCE;
	}
}
```

PMD reports a violation at line ..., but that's wrong. That's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Maven plugin 3.24.0
